app/auth/services_views.py

```diff
@@ -0,0 +1,93 @@
+import typing as tp
+
+import jwt
+
+from datetime import timedelta, datetime
+
+from settings import (
+    SECRET_KEY,
+    ACCESS_TOKEN_EXPIRE_MINUTES,
+    ACCESS_TOKEN_JWT_SUBJECT,
+    TOKEN_ALGORITHM,
+    REFRESH_TOKEN_EXPIRE_MINUTES,
+    REFRESH_TOKEN_JWT_SUBJECT,
+    DEFAULT_ACCESS_TOKEN_EXPIRE_MINUTES,
+    DEFAULT_REFRESH_TOKEN_EXPIRE_MINUTES,
+)
+from utils.exceptions import SerializerValidationError
+from utils.serializers import serializer_data_type
+
+
+def create_tokens(data: serializer_data_type) -> tp.Dict[str, str]:
+    """
+    Create tokens
+    Args:
+        data: dict with username and password
+    Return:
+        dict with access and refresh token
+    """
+
+    access_token = create_token(token_type='access', data=data.copy())
+    refresh_token = create_token(token_type='refresh', data=data.copy())
+    return_data = ({'refresh_token': refresh_token, 'access_token': access_token})
+
+    return return_data
+
+
+def create_token(token_type: str, data: serializer_data_type) -> str:
+    """
+    Create access token using credentials or refresh token
+    Args:
+        token_type: access or refresh
+        data: user credentials
+    Returns:
+        new token
+    """
+
+    helping_data: tp.Any = {
+        'refresh': {
+            'expire': REFRESH_TOKEN_EXPIRE_MINUTES,
+            'default_expire': DEFAULT_REFRESH_TOKEN_EXPIRE_MINUTES,
+            'subject': REFRESH_TOKEN_JWT_SUBJECT
+        },
+        'access': {
+            'expire': ACCESS_TOKEN_EXPIRE_MINUTES,
+            'default_expire': DEFAULT_ACCESS_TOKEN_EXPIRE_MINUTES,
+            'subject': ACCESS_TOKEN_JWT_SUBJECT
+        }
+    }
+
+    if helping_data[token_type]['expire'] is not None:
+        helping_data['expire'] = datetime.utcnow() + timedelta(minutes=float(helping_data[token_type]['expire']))
```

Comment:
why save it to the dict if on line 67 you just get it back?

Reply:
done